Makefile

```diff
@@ -50,7 +61,11 @@ LISTSDIR = $(MKFILES_DIR)lists/
 # generated folders
 
 #! The directory for object assembly files (stores `.o` files)
+ifdef __EMSCRIPTEN__
+OBJDIR = ./obj/emscripten_obj/
```

Comment:
Yes, it only ensures there won't be collision between the .o files
calling `make clean` works when you manually encounter the error, but then it means scripts can't use `make` themselves without calling `make clean` anymore, so IMO having a separate BINDIR makes sense.

Another drawback of having a separate BINDIR is that you must both `make clean && emmake clean` to actually clean every .o . Or you can just be a barbarian like me and `rm -rf bin obj` instead of relying on the makefile to clean.
Actually, I don't really understand why the clean rules don't jurt `rm -rf obj/`, but that's another topic

You probably have a more educated opinion than I do, so I'll go with what you say

Reply:
Well, if you do `make help`, you can see that there do exist rules that directly `rm -rf` the ./obj & ./bin folders - they are just not placed directly inside build.mk:
```sh
./mkfile/rules/clean.mk
make clean                    #! Deletes files generated by the `make all` default rule
make clean-all                #! Deletes every generated file/folder
make clean-obj                #! Deletes the ./obj folder
make clean-bin                #! Deletes the ./bin folder
make clean-log                #! Deletes the ./log folder
```
So yeah, these rules do exist - but, admittedly, `rm -rf obj` has fewer characters to type out than `make clean-obj` ^^

I really believe it is best to accept these name collisions, it is coherent with the current build system: each `OSMODE` has `.o` files which have name collisions with each other. Since `emscripten` is essentially a new `OSMODE`, I think it is fine to assume that devs need to know that a `make clean` is needed when switching between OSMODE targets.

Perhaps in the future, we will need to make everything distinct (so, various `$(OSMODE)` folders inside the `./obj` folder - perhaps also use `debug`/`release` subfolders as well) - But I hvaven't found a compelling enough use-case which requires this yet.

Reply:
Okay, will fix this next commit